rust/models/src/host_info.rs

```diff
@@ -4,29 +4,87 @@
 
 use std::collections::HashMap;
 
+use crate::Host;
+
 /// Information about hosts of a running scan
 #[derive(Debug, Clone, Default, PartialEq, Eq)]
 #[cfg_attr(
     feature = "serde_support",
     derive(serde::Serialize, serde::Deserialize)
 )]
 pub struct HostInfo {
-    /// Number of all hosts, that are contained in a target
-    pub all: u32,
-    /// Number of hosts, that are excluded from the target
-    pub excluded: u32,
-    /// Number of hosts, that are not reachable (alive-test failed)
-    pub dead: u32,
-    /// Number of hosts, that are reachable (alive-test succeeded)
-    pub alive: u32,
-    /// Number of hosts, that are currently queued for scanning
-    pub queued: u32,
-    /// Number of hosts, that are already finished scanning
-    pub finished: u32,
-    #[cfg_attr(
-        feature = "serde_support",
-        serde(skip_serializing_if = "Option::is_none")
-    )]
-    /// IPs of hosts, that are currently scanned.
-    pub scanning: Option<HashMap<String, i32>>,
+    all: u64,
+    excluded: u64,
+    dead: u64,
+    alive: u64,
+    queued: u64,
+    finished: u64,
+    // Hosts that are currently being scanned. The second entry is the number of
+    // remaining VTs for this host.
+    scanning: HashMap<String, usize>,
+}
+
+impl HostInfo {
+    pub fn new(all: u64, excluded: u64, dead: u64, alive: u64, queued: u64, finished: u64) -> Self {
```

Comment:
feel free to ignore:

I know that a lot of people prefer constructor methods over a constructor but for this amount of parameter I would prefer reading:
```
HostInfo { 
    all: 1,
    excluded: 1,
    .. Default::default(),
}
```
over 
```
 HostInfor::new(1, 1, 0, 0, 0, 0)
```